core/src/main/java/de/danoeh/antennapod/core/service/playback/PlaybackService.java

```diff
@@ -1854,6 +1876,12 @@ public void onSeekTo(long pos) {
         @Override
         public void onSetPlaybackSpeed(float speed) {
             Log.d(TAG, "onSetPlaybackSpeed()");
+            PlaybackPreferences.setCurrentlyPlayingTemporaryPlaybackSpeed(speed);
+            if (currentMediaType == MediaType.VIDEO) {
+                UserPreferences.setVideoPlaybackSpeed(speed);
+            } else {
+                UserPreferences.setPlaybackSpeed(speed);
+            }
```

Comment:
This copied from PlaybackController, right? Would it make sense to move the code to `setSpeed`? Having multiple copies of the same code will probably cause issues later, otherwise.

Reply:
Yeah, that would make sense. I think my worry is that I'm still not sure where SpeedChangedEvents from the PlaybackController are handled, so that might mess something up down the line.